### ISSUE TITLE
Tech team summaries

### ISSUE THREAD

**Issue by kdembler** [Joystream/atlas#815]
This is a long-living issue for the Atlas tech team to post summaries of their work to share with the community

**Comment by kdembler** [comment]
# 14.06.21 - 18.06.21

### What the team worked on:

- Introduced style linting
- Introduced global state management
- Improved handling of notifications
- Updated form validation library
- Improved font imports, fixing a long standing issue
- Made tweaks to video tabs to improve UX
- Introduced centralized logger
- Various improvements to handling uploads
- Introduced dialog to let users know their browser is outdated
- Added possibility to dynamically update the featured hero video
- Smaller UI tweaks
- Various DX improvements & updates

### Opens:

- @bedeho last week I've touched upon this and I think we need to start thinking about how to connect Orion data with query node data to enable rich querying and features like "popular". We have this issue opened for some time - https://github.com/Joystream/atlas/issues/132 - I've added a comment there to describe what I think we could do about this. (tldr: introduce a "gateway node" that will connect data from query node and orion). We should discuss it soon
- I've added 2 comments regarding use of labels and didn't receive a response https://github.com/Joystream/atlas/issues/777

**Comment by kdembler** [comment]
# 21.06.21 - 25.06.21

### What the team worked on:

- Moving various pieces of app to use global state management
- Introduction of e2e tests
- Further improvements to uploads handling (now query node is the truth, we don't rely on local state as much)
- Refactor of transaction handling to make it bullet proof and fix long standing issue related to publishing videos/creating channels
- Refactored handling of assets to open possibility of additional features in the future (it's now centralized so if we wanted to include retries, we just need to do it in one place)
- Fixes for some problems with overlays
- Small tweaks & bug fixes
- Some QoL improvements (e.g. persisting player volume between sessions)
- Further DX improvements

**Comment by eldiegod** [comment]
# 28.06.21 - 02.07.21

### What the team worked on:

- Refactored how we handled video drafts in a more robust way that uses global state management
- Kept working on adding more e2e testing tho we hit a roadblock making signing transactions work on the tests and decided to deprioritize for now while we work on the viewer side updates
- Made a playground to test out the loading animations handed by the designers
- We added an option to export local state (everything saved in local storage) as a JSON (helpful for debugging)
- Fixes for the video grid that in certain resolutions would vary sizes
- Other small fixes and refactors as part of our maintenance sprint
- Added a timeout to notifications about network connectivity and improved copy
- Fixed a bug that when opening a dialog while another was open led to breaking the new dialog's layout 
- Improve asset upload notifications to be more ergonomic and easier to digest

**Comment by eldiegod** [comment]
# 05.07.21 - 09.07.21

### What the team worked on:

- Added ability to toggle full-screen on pressing F key
- Added ability to move video 5 and 10 seconds forward and backward with keyboard
- Added ability to mute, increase/decrease volume with keyboard
- Orion query to get videos and channels with the most views by given period
- Visual update for the banner component
- Updated the color palette
- Button component visual update
- Play / Pause controls and timeline bar visual update
- Volume controls visual update and now smoother
- Channel cover component visual update
- Channel views new layout
- Sorting functionality for the channel view
- Pagination for videos on the channel view 
- Added tabs for the channel view

### Milestones progress in days left (approx)

- Channel view update 1.5 / 2 days (14.07.21)
- Video content pages  4.3 days (17.07.21)
- Video player update 2.5 days + hard to estimate task (>15.07.21)

**Comment by kdembler** [comment]
# 12.07.21 - 16.07.21

### What the team worked on:

Not exactly sure as I was away, but in general the team continued work on set milestones.

### Milestones estimates

- Channel view update (@eldiegod) - end of week (23.07)
- Video player update (@drillprop) - end of week (23.07)
- Video content pages (@wyrdrender) - end of next week (30.07)
- Misc hotfixes (@kdembler) (asset fallbacks, asset monitoring, better errors for video player) - end of week (23.07)

**Comment by kdembler** [comment]
# 19.07.21 - 23.07.21

### What the team worked on:

- Continued work on channel view update (details, search)
- Continued work on player update (different player overlays, tooltips, keyboard shortcuts)
- Continued work on video content pages (Orion improvements, work on presentation components, view layouts)
- Production hotfixes (mostly asset-related fixes, retries, timeouts, etc)

### Milestones estimates

- Channel view update (@eldiegod) - Wednesday (28.07), delayed because of some bugs found and overall slower progress of review
- Video player update (@drillprop) - Wednesday (23.07), delayed mostly because of scope increase (compact player, ending screen with next video, autoplay) and need to reimplement some of the player UI manually
- Video content pages (@wyrdrender) - likely middle of next week (4.08)
- Misc hotfixes (@kdembler) (error recording/reporting) - depends on exact scope, likely Wednesday/Thursday (28,29.07)

**Comment by kdembler** [comment]
# 19.07.21 - 23.07.21

### What the team worked on:

- Last tweaks for the channel update
- Wrapped up player update
- Continued work on video content pages (Orion improvements, work on presentation components, view layouts)
- Atlas error recording service
- Some fixes for prod (asset-related)

### Milestones estimates

- Channel view update (@eldiegod) - At this point only review left, will probably be merged by the end of Monday (02.08)
- Video player update (@drillprop) - finished
- Video content pages (@wyrdrender) - hard to say, still depends on outcome of discussion around what to do with some "discover" sections. Work is progressing slower than anticipated, we will likely pull in additional dev to that sprint to help out a bit. Assuming 2 devs, I'm hoping for the end of next week (06.08), but may stretch out a bit longer
- Atlas error recording - will be wrapped up by the end of the next week (06.08)

**Comment by kdembler** [comment]
# 02.08.21 - 06.08.21

### What the team worked on:

- Bug fixes and improvements for channel view update
- Bug fixes and tweaks for video player
- Continued work on video content pages (new Orion capabilities, new video&channel sections)
- Atlas error recording service
- Misc improvements - consistency with design, renaming components, skeleton loader, etc

### Milestones estimates

- Channel view update (@eldiegod) - **Finished**, will need design review and some tweaks but will be deployed today
- Video content pages (@wyrdrender, @drillprop) - Still in progress, requirements shifted a bit at the start of the week, we needed additional changes on Orion side. My current best guess is **end of the next week (13.08)** for initial release and half a week for tweaks in production
- Atlas error recording  (@kdembler) - Slightly delayed because of more complexities in the ELK service than anticipated and my reduced availability this week. I aim to wrap up by **Wednesday (11.08)**
- Responsive video grid (@eldiegod) - Just started, should be wrapped by the **end of the next week (13.08)**

**Comment by kdembler** [comment]
# 09.08.21 - 13.08.21

### What the team worked on:

- Continued work on video content pages (views were finished, review feedback improvements)
- Bug fixes, improvements and release of channel view update
- Implementation of responsive layout grid, introduction to channel view
- Bug fixes and tweaks for video player
- Clean up of logs sent to Sentry, tracking service work
- Misc improvements - fixes for some problems that were fixed in the design but never implemented, and others

### Milestones estimates

- Channel view update (@eldiegod) - **Closed**, needed some time this week for issues we've found
- Video content pages (@wyrdrender, @drillprop) - Review and tweaks left, should be done by **end of Tuesday (17.08)**
- Atlas error recording  (@kdembler) - Scope here was increased during work. I've discovered that most of the errors we sent to Sentry are not really great and even when error is captured, we have hard time figuring what actually happened. So as part of the sprint I'm also refactoring what we send to Sentry so we can get full picture. Based on my engagement in other things, I expect to finish by **end of next week (20.08)**
- Responsive grid (@eldiegod) - Mostly done, one issue left, should be done by **end of Tuesday (17.08)**

**Comment by bedeho** [comment]
# Meeting Summary: Gateway Node

Attendants

- Klaudiusz
- Bedeho

## Background

There is an increasing list of enhancements to atlas that may warrant the introduction of a proto-Gateway node, well in advance of the core access financing function that Gateways are supposed to serve. The prupose of the meeting was to get an overview of these enhancements, as well as decide how to proceed in terms of timing and scope of this work.

## Proposed Features

- pre-processing of assets
 - transcoding
 - video preview images during scrubbing
- richness of the data we can query from atlas: data split, causes complexity, and restricts acutal options.
- aggregations.
- error recording & analytics
- private client side user state : who you follow, etc., so we get cross session & browser state, here we could also consider introducing user accounts, with user-name & passwords, etc.
- featured content
- DMCA information
- terms of service & licensing
- media sharing previews.
- make a docker image for atlas.
- stateless stitching layer: primarily to simplify program development, possibly caching layer, 

## Observations

Many of these are very useful, some are nice to have, but in terms of timing its hard to prioritize any major systematic effort to introduce these right now. Specifically features which have to do with rationalizing how to query across the chain, follower data and consumption data, requires resolving how the shared data layer for this should be organised, and his has not been settled yet, so we can't really execute on this yet.

## Conclusion

Klaudiusz will try to tackle some core issues here over time, in a piecemeal way, when other major sprints are not occupying him.

**Comment by kdembler** [comment]
# 16.08.21 - 20.08.21

### What the team worked on:

- Review work for video content pages + changes to Orion
- Fixes for responsive grid in channel view
- Update of video hero component
- Continued work on error capture refactor & logging asset metrics
- Start of Discovery view work
- Various misc tweaks&fixes

### Milestones estimates

- Video content pages (@wyrdrender, @drillprop) - **Almost done**, will likely be deployed today or by the end of the weekend. We've noticed one issue with Orion that needed to be solved before release. I know Kuba has some feedback regarding design implementation so we may need to spend some time on tweaks next week.
- Atlas error recording  (@kdembler) - Almost done, error improvements & asset metrics were already merged, will be deployed alongside video content pages. Two things left to do: capturing upload related events and also creating some publicly visible dashboard. I hope for this to be done by **Tuesday (24.08)**
- Responsive grid (@eldiegod) - **Done**
- Discovery page (@eldiegod) - Started, but the scope is still not fully defined so hard to estimate at this point

**Comment by kdembler** [comment]
# 23.08.21 - 27.08.21

### What the team worked on:

- Post-feedback fixes for video content pages
- Wrap up of asset metrics logging (dashboard available at https://orion.joystream.org/sp-dashboard)
- Upload flow sprint kick-off and implementation (tracking upload finalization by query node)
- Work on Discover page
- A lot of tweaks&fixes (updated breakpoints, responsive helpers, skeleton loaders for pending images, etc)

### Milestones estimates

- Video content pages (@wyrdrender) - Only one point from feedback left, already implemented but in review, should be done on **Monday**
- Atlas error recording  (@kdembler) - **Done**
- Discovery page (@eldiegod) - Some work already done, but not all of the work has been estimated so far. I estimate **08.09** is the earliest valid date for completion
- Upload flow improvements (@drillprop) - Issues have been created (not on github yet), some work has already been done. Hard to estimate at this point but I think at least **2 weeks of work (10.09)**

**Comment by kdembler** [comment]
# 30.08 - 03.09

### What the team worked on:

- Post-feedback fixes for channel view
- Work on Discover sprint - category cards, featured categories, page layout
- Work on upload flow improvements sprint - continued tracking of upload completion, improvements for the video workspace
- Migration to `vite` bundler - much faster dev experience, 65% smaller bundle size when opening the front page (faster app download)
- Tweaks&fixes

### Milestones estimates

- Video content pages (@wyrdrender) - **Done**
- Discovery page (@eldiegod) - Some progress was made but Diego will be away for the next week. I expect **17.09** as earliest completion date
- Upload flow improvements (@drillprop) - Good progress was made, I think it's possible to wrap up by the **end of next week (10.09)**, possibly a bit longer
- DX improvements - (@kdembler) - one small task I'd like to tackle left, should be done by **end of Monday (06.09)**

### Absences
Both Diego and Rafał will be away next week

**Comment by kdembler** [comment]
# 06.09 - 10.09

### What the team worked on:

- Work on upload flow improvements sprint - new video tile, further video workspace tweaks, uploads manager improvements
- DX improvements - admin overlay to easily navigate and switch environments when developing
- Backlog of PRs for review
- Tweaks&fixes

### Milestones estimates

- Discovery page (@eldiegod) - Still expect 17.09 as earliest completion date, more likely **22.09** including reviews
- Upload flow improvements (@drillprop) - After some changes in the design, this will likely take a bit longer than originally anticipated. My current guess is 17.09 for initial work finish and **22.09** including the review
- DX improvements - (@kdembler) - **Done**

### Absences
Klaudiusz will be away next week

**Comment by kdembler** [comment]
# 20.09 - 24.09

### What the team worked on:

- Final tasks were finished for the upload flow sprint. Almost everything is ready by now, just reviews left
- Continued work on the discovery sprint - all videos section, filtering bar, some new shared components

### Milestones estimates

- Discovery page (@eldiegod, @kdembler) - Still going, I expect **end of next week (1.10)** although I think we will need to cut the scope a bit. Will share more during call on Monday
- Upload flow improvements (@drillprop, @wyrdrender) - Almost all the work is already done but the reviewing (mostly me) is the bottleneck. If nothing major comes up I think we may be able to merge by **Wednesday (29.09)**

**Comment by kdembler** [comment]
# 27.09 - 01.10

### What the team worked on:

- Some review followups for the upload flow sprint. The work has been finished
- Further work on the discovery sprint. The scope was expanded a bit to allow the community to manage the featured content. We also run into some small issues with component refactors that are part of this sprint. Other than we finished the main layout for video category page, introduced a background video player and made improvements to the video hero component.
- Work on the search sprint has been started. After kickoff and issue creation, we implemented bottom navigation bar for mobile and updated the top navigation.

### Milestones estimates

- Upload flow improvements (@drillprop, @wyrdrender) - **Done?** We finished and merged the work for the current scope but there are still some design issues open in the milestone so seems there will still be some work done here?
- Discovery (@eldiegod, @drillprop) - Still going, unfortunately with a bit more delay. I think we should finish by the **end of the next week (08.10)** although that also depends on when we will get the videos to be featured from the community. Seems this may be a blocker
- Search sprint (@wyrdrender) - The progress has been good so far, based on the current estimates I think we should finish by  **13.10**

### Topics for Monday meeting

- Stale `Discover new channels` section on Home page. We should either implement the missing query node features to enable more rich queries or change the policy for picking those.
- Let's discuss the community-managed content in Atlas a bit.

**Comment by kdembler** [comment]
# 04.10 - 08.10

### What the team worked on:

- Additional improvements to upload flow sprint after new designs have been prepared
- Continued work on the discovery sprint. We're done with the video hero section for the category view and almost done with filtering bar.
- Further work on search sprint. Updates to topbar were tackled as well as changes to the search component.

### Milestones estimates

- Upload flow improvements (@drillprop) - Last updates being done here. I'm confident we can deploy by Wednesday at latest.
- Discovery (@eldiegod) - We need to wrap up the filters bar and add an infinite video grid to the view, but those seem like smaller tasks. The bigger part of this will be the agreed changes to Orion for it to handle featured content as well. So we will need to finish those and also add support to Atlas. I think we could wrap up by the end of the next week but I'm not sure. Even if we don't, Diego will likely be able to transition to something else.
- Search sprint (@wyrdrender) - Steady progress, although I think it may take a bit longer than the initial estimate. It's also a bit blocked by the filters bar done as part of the discovery sprint. End of the next week seem to be a good guess

**Comment by kdembler** [comment]
# 11.10 - 15.10

### What the team worked on:

- Implementation of final improvements for upload flow sprint. This included some newly designed variants for components and some final tweaks.
- Continuation of discovery sprint work. Unfortunately wrapping up of the filtering bar took some time. We also have the necessary changes to the infinite grid ready and being reviewed.
- As part of the search sprint, wrapping up the searchbox and implemented animations and keyboard shortcuts for it.
- Creation of process for featured content updates by community and necessary changes to Orion
- Started introduction of design tokens and color updates.

### Milestones estimates

- Upload flow improvements - **Done** 🎉
- Discovery (@eldiegod) - Filters bar is done and infinite grid is almost ready. Other than we only need featured content changes introduced to Atlas and community actually setting all the featured content. This creates some uncertainty about wrapping this milestone up, I think **end of next week** may be a good guess at the time.
- Search sprint (@wyrdrender) - This week we hit a bit of a delay with necessary changes to the searchbox. I think at the moment a good estimate would by **end of next week** as well
- Design tokens (@drillprop started, most likely @kdembler will continue, we'll see) - We started codegen for design tokens in the app and started with the color updates. This should be also finished by the **end of next week**

### Absences

@drillprop will be away for 2 weeks, back on 2nd of November

**Comment by kdembler** [comment]
# 18.10 - 22.10

### What the team worked on:

- Continuation of discovery sprint work. This included final tweaks to the page layout and infinite grid and also adding support to fetch featured content from Orion, we're currently wrapping up.
- As part of the search sprint, work on the results page and tweaks for searchbox animations
- Continued work on design tokens and color updates.

### Milestones estimates

- Discovery (@eldiegod) - We're just wrapping up the work, although the release of this will be dependant on the community to provide featured content. Development will likely be done on **Monday**
- Search sprint (@wyrdrender) - Because of Rafał's unexpected leave last week this has to be slightly delayed. Also I think design review may spawn some new issues to be tackled so safe bet is likely **end of next week (29.10)**
- Design tokens (@kdembler) - Not much progress was done because of my absence and focusing on reviews of other work. We're still figuring out answers to some of the problems with the design team so this isn't the main priority. For now maybe let's say **end of next week (29.10)** as the estimate, but this may shift along with priorities.

### Absences

@drillprop back on 2nd of November

**Comment by kdembler** [comment]
# 25.10 - 29.10

### What the team worked on:

- Wrapping up of the search sprint. We've hit up a couple of bigger issues to make it work properly on mobile (mostly iOS)
- Wrapping up discovery sprint. This included support for dynamic featured content coming from Orion, some changes to Orion itself and figuring out what to do with video hero once we get rid of highres thumbnails
- Some work on design tokens in the meantime
- Started preparation on Giza scope

### Milestones estimates

- Discovery (@eldiegod) - We have single visual feature to tackle (sticky header) and then this will be blocked by the community providing the content. Engineering work should be finished by **Tuesday**.
- Search sprint (@wyrdrender) - All done, just final testing left, can be considered finished.
- Design tokens (???) - Almost no progress was made this week as the full team was occupied with other things, probably Bartosz will pick it up once he comes back from his vacation.

### Absences

@drillprop back on 2nd of November.

All Polish team members will be absent on Monday.

**Comment by kdembler** [comment]
# 01.11 - 05.11

### What the team worked on:

- Last testing, review and tweaks for the search sprint
- Bit of work on sticky header in the video category view
- Work on design tokens and colors updates
- Further research into Giza
- Update to dialogs/modals to conform with the newest designs
- Work to export icons directly from Figma API
- Work on cleaning up the component structure in the codebase
- Work on moving the data stitching functionality to Orion

### Milestones estimates

- Discovery (@eldiegod) - **Done**, waiting for the content from community
- Search sprint (@wyrdrender) - **Done**, although an issue came up that was brought up internally with the team. Release will depend on the decision we make about the way forward
- Design tokens (@drillprop) - **Done**
- Other than sprints above, we started 3 smaller pieces of work that possibly don't deserve their own sprints (Figma icon export, cleaning up component structure, moving stitching to Orion)

### Absences

All Polish team members will be absent on Thursday, some of them also on Friday, details here: https://docs.google.com/spreadsheets/d/1UE503IGKP1JucsI47PR2z28bvuRKikDMvFf-VvMqdOI/edit#gid=0

**Comment by kdembler** [comment]
# 08.11 - 12.11

### What the team worked on:

- Giza planning
- Exporting icons directly from Figma API
- Cleaning up the component structure in the codebase, cleaning up Storybook
- Moving the data stitching functionality to Orion
- Bunch of smaller tweaks and fixes

### Milestones estimates

- Discovery - waiting for the content from community
- Other than that, we're still working on cleaning up Storybook and moving stitching to Orion

**Comment by kdembler** [comment]
# 15.11 - 19.11

### What the team worked on:

- Continued exporting icons directly from Figma API (done)
- Continued cleaning up the component structure (done)
- Continued cleaning up Storybook (done)
- Continued on moving the data stitching functionality to Orion
- Introduced updated Typography tokens
- Started Giza work
- Added support for 2 session recording tools
- Work on improving the docs (readme, issue templates, etc)
- Investigating performance issues with some of the components, this is mostly to understand the mistakes we have made so we don't make them in the future
- Introduced experimental embedded player view
- Bunch of smaller tweaks and fixes

### Milestones estimates

- Discovery (@kdembler on the community side) - waiting for the content from community. Seems to be pretty close, I know uploads have started, should release next week.
 - Giza (@kdembler) - still hard to tell, I'm still figuring some parts out. We should have some estimate next week. At this point I suppose end of November may be a good guess.
 - Misc tweaks
   - Moving data stitching to Orion (@drillprop, @kdembler) - main work has been done, we need to review but I didn't have enough time to do it this week, it's on the backburner atm
   - Typography tokens (@drillprop) - we've successfully introduced all the new tokens, there's little work left to use them in the app and make necessary changes to some of the components - should be wrapped up mid next week
   - Performance investigation (@wyrdrender) - we likely need few days to fully understand some of the issues, we likely won't do too much work right away

### Planned absences

Bartosz will be absent on Monday

**Comment by kdembler** [comment]
# 22.11 - 26.11

### What the team worked on:

- Fixes for Discovery bugs discovered once community set the featured content (done)
- Using the new typography (with new fontface) and updating the app to work properly with it
- Continued on moving the data stitching functionality to Orion
- Continued Giza work (not much progress due to blockers)
- Made changes to the session recording tools, introduced a user feedback tool
- Performance fixes for some of the components
- Bunch of smaller tweaks and fixes
- Some develop QoL improvements

### Milestones estimates

- Discovery - done
- Giza (@kdembler) - I'm unsure whether we are fully unblocked already, but if yes I'd like to direct my full attention towards this this week. I think we should be able to have a WIP version by the end of the week and nail it down by the end of the next week.
- Misc tweaks
   - Moving data stitching to Orion (@drillprop, @kdembler) - we started reviewing this week and also made necessary changes to Atlas. I think we should be able to wrap up in this week.
   - Typography changes (@eldiegod) - almost all the changes were made and a PR is open, though review will likely bring up some followups
   - Performance investigation (@wyrdrender) - We should finish Monday/Tuesday

### Planned absences

Bartosz will be absent on Monday

**Comment by kdembler** [comment]
We changed the way we work - no longer needed
